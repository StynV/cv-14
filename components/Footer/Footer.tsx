import Image from 'next/image'
import Logo from "@/public/images/Github.png"
import styles from './footer.module.scss'

const Footer = () => (
    <footer className={styles.footer}>
        <a
            href="https://github.com/StynV/cv"
            target="_blank"
            rel="noreferrer"
            itemProp='author'
        >
            <Image
                src={Logo}
                alt={'Source code'}
                width={80}
                height={80}
            />
        </a>
    </footer>
)

export default Footer
