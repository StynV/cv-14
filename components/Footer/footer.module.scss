@import "../../styles/vars.scss";

.footer {
  display: flex;
  justify-content: center;
  border-top: 1px solid lightgray;
  
  @media (min-width: $break-lg) {
    margin-left: calc(#{$spacing-bace} * 20);
    margin-right: calc(#{$spacing-bace} * 20);
  }
}