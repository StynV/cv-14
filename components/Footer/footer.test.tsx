import { render, screen } from '@testing-library/react'
import Footer from './Footer'

describe('Footer', () => {
  it('renders a Footer', () => {
    const { container, getByText } = render(<Footer />)

    expect(container).toMatchSnapshot()

    expect(container.querySelector('footer')).toHaveClass('footer')

    expect(container.querySelector('a[href="https://github.com/StynV/cv"]')).toBeVisible()
    expect(container.querySelector('img')).toBeVisible()
  })
})