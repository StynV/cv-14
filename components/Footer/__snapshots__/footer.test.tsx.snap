// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders a Footer 1`] = `
<div>
  <footer
    class="footer"
  >
    <a
      href="https://github.com/StynV/cv"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Source code"
        data-nimg="1"
        decoding="async"
        height="80"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2FGithub.png&w=256&q=75"
        srcset="/_next/image?url=%2Fimages%2FGithub.png&w=96&q=75 1x, /_next/image?url=%2Fimages%2FGithub.png&w=256&q=75 2x"
        style="color: transparent;"
        width="80"
      />
    </a>
  </footer>
</div>
`;
